torch/fx/experimental/symbolic_shapes.py

```diff
@@ -186,79 +177,225 @@ def magic_impl(self, other):
             if isinstance(other, PySymInt):
                 other = other.expr
             # TODO: consider constant prop here
-            return PySymInt(func(self.expr, other), self.shape_env)
+            expr = self.shape_env.replace(self.expr)
+            other = self.shape_env.replace(other)
+            out = func(expr, other)
+            out = self.shape_env.replace(out)
```

Comment:
How come this replace call is necessary? Replace operates on free variables, and func should never introduce new free variables, so replacing on the inputs should be sufficient

Reply:
I mainly just use it as a replacement for expand, but sure, we could also remove it :)

Reply:
well, expanding after the operation is legit, but calling it a replace is very unclear lol